src/services/chirpstack/chirpstack-gateway.service.ts

```diff
@@ -58,31 +66,72 @@ export class ChirpstackGatewayService extends GenericChirpstackConfigurationServ
 
         gateway.organization = await this.organizationService.findById(dto.organizationId);
 
-        const result = await this.post("gateways", dto);
-        await this.gatewayRepository.save(gateway);
-        return this.handlePossibleError(result, dto);
+        const req = new CreateGatewayRequest();
+        const location = new Location();
+        this.mapToLocationChirpstack(location, dto);
+
+        const gatewayCs = new GatewayCs();
+        await this.mapToGatewayChirpstack(gatewayCs, dto, location);
+        Object.entries(dto.gateway.tags).forEach(([key, value]) => {
+            gatewayCs.getTagsMap().set(key, value);
+        });
+
+        req.setGateway(gatewayCs);
+        try {
+            await this.gatewayRepository.save(gateway);
+            await this.post("gateways", this.gatewayClient, req);
+            return { success: true };
+        } catch (e) {
+            this.logger.error(`Error from Chirpstack: '${JSON.stringify(dto)}', got response: ${JSON.stringify(e)}`);
+            throw new BadRequestException({
+                success: false,
+                error: e,
+            });
+        }
+    }
+
+    async mapToGatewayChirpstack(
+        gateway: GatewayCs,
+        dto: CreateGatewayDto | UpdateGatewayDto,
+        location: Location,
+        gatewayId?: string
+    ) {
+        gateway.setGatewayId(gatewayId ? gatewayId : dto.gateway.gatewayId);
+        gateway.setDescription(dto.gateway.description);
+        gateway.setName(dto.gateway.name);
+        gateway.setLocation(location);
+        gateway.setStatsInterval(30);
+        gateway.setTenantId(dto.gateway.tenantId ? dto.gateway.tenantId : await this.getDefaultOrganizationId());
+    }
+    mapToLocationChirpstack(location: Location, dto: CreateGatewayDto | UpdateGatewayDto) {
+        location.setAccuracy(dto.gateway.location.accuracy);
+        location.setAltitude(dto.gateway.location.altitude);
+        location.setLatitude(dto.gateway.location.latitude);
+        location.setLongitude(dto.gateway.location.longitude);
+        location.setSource(dto.gateway.location.source);
     }
 
-    addUserToTags(dto: CreateGatewayDto, userId: number): { [id: string]: string | number } {
+    addUserToTags(dto: CreateGatewayDto, userId: number): { [id: string]: string } {
         const tags = dto.gateway.tags;
         tags[this.CREATED_BY_KEY] = `${userId}`;
         tags[this.UPDATED_BY_KEY] = `${userId}`;
         return tags;
     }
 
-    updateUpdatedByTag(dto: UpdateGatewayDto, userId: number): { [id: string]: string | number } {
+    updateTags(dto: UpdateGatewayDto, userId: number): { [id: string]: string } {
         const tags = dto.gateway.tags;
         tags[this.UPDATED_BY_KEY] = `${userId}`;
+        tags[this.CREATED_BY_KEY] = `${dto.gateway.createdBy}`;
```

Comment:
If i'm not setting the created-by tag, it will be erased in the chirpstack tags. I know that we are adding these tags and data to the database now - so should we still add the tags to chirpstack? If yes - then we have to set created-by. If no, then I guess we can just erase the function :-)